modules/network/src/main/java/org/apache/ignite/network/scalecube/ScaleCubeTopologyService.java

```diff
@@ -54,37 +52,16 @@ void setLocalMember(Member member) {
     void onMembershipEvent(MembershipEvent event) {
         ClusterNode member = fromMember(event.member());
 
-        String memberName = member.name();
-
-        switch (event.type()) {
-            case ADDED:
-                members.put(memberName, member);
-
-                fireAppearedEvent(member);
-
-                break;
-
-            case LEAVING:
-                members.remove(memberName);
-
-                fireDisappearedEvent(member);
+        if (event.isAdded()) {
```

Comment:
what about `LEAVING` event?

Reply:
LEAVING event makes sense only if we want to let the user know the node has left intentionally - similar to NODE_LEFT event in ignite2.
Currently I see no demand in such functionality. It may be added later, if needed.

Reply:
Ok, why not